src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +25,26 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
+        StringBuilder data = new StringBuilder();
+        try {
+            List<String> lines = Files.readAllLines(Paths.get(fileName));
+            for (String s : lines) {
+                data.append(s);
+            }
```

Comment:
you can find a way to iterate over each character already at this point (read per each char)
then, using a method in class Character, you can check if the specific char suits you
and add it to the set

Reply:
understood, found, refactored